techtv2ovs/utils.py

```diff
@@ -0,0 +1,302 @@
+
+
+def get_owner(email):
+    """
+    Get/create a user for emails ending with `@mit.edu`, otherwise return a default user
+
+    Args:
+        email (str): An email address
+
+    Returns:
+        User: A User object.
+    """
+    if email and email.endswith('@mit.edu'):
+        user, _ = User.objects.get_or_create(username=email, defaults={
+            'email': email
+        })
+        return user
+    return User.objects.get(username=settings.LECTURE_CAPTURE_USER)
+
+
+def remove_tags(txt):
+    """
+    Replace any HTML tags with spaces, HTML-encoded characters with unicode
+
+    Args:
+        txt (str): Text to remove HTML tags from.
+
+    Returns:
+        str: String with HTML tags/characters removed.
+    """
+    if not txt:
+        return ''
+    return normalize("NFKD", unescape(strip_tags(txt)))
+
+
+class TechTVImporter(object):
+    """
+    Class for handling TechTV to OVS imports
+    """
+
+    def __init__(self, **kwargs):
+        """
+        Assign class properties based on kwargs
+        """
+        self.db_user = kwargs.get('db_user', None)
+        self.db_pw = kwargs.get('db_pw', None)
+        self.db_name = kwargs.get('db_name', 'techtv')
+        self.db_host = kwargs.get('db_host', 'localhost')
+        self.collection = kwargs.get('collection', None)
+        self.aws = kwargs.get('aws', False)
+        self.protected = kwargs.get('protected', 1)
+        self.output = kwargs.get('output', sys.stdout)
+
+    def run(self):
+        """
+        Run the import
+        """
+        with MySQLdb.connect(
+            user=self.db_user, password=self.db_pw, db=self.db_name, host=self.db_host
+        ) as self.connection:
+            bonobo.settings.LOGGING_LEVEL.set(logging.ERROR)
+            bonobo.run(
+                self.etl_graph()
+            )
+
+    def process_thumbs(self, ttv_video):
+        """
+        Copy over all 'jumbo' thumbnails from TechTV to OVS buckets for a TechTVVideo.
+        Create a VideoThumbnail object for each.
+
+        Args:
+            ttv_video (TechTVVideo): The TechTVVideo to process
+        """
+        ttv_video.thumbnail_status = ImportStatus.CREATED
+        s3client = boto3.client('s3')
+        s3list = s3client.list_objects_v2(Bucket=TTV_THUMB_BUCKET, Prefix=ttv_video.external_id)
+        if 'Contents' not in s3list or not s3list['Contents']:
+            ttv_video.thumbnail_status = ImportStatus.MISSING
+            ttv_video.save()
+            return
+        user_list = [thumb for thumb in s3list['Contents'] if '{}/jumbo'.format(ttv_video.ttv_id) in thumb['Key']]
+        default_list = [thumb for thumb in s3list['Contents'] if 'thumbnails/jumbo' in thumb['Key']]
+        idx = 0
+        for thumb in user_list + default_list:
+            new_key = thumb['Key'].split('/')[-1].replace('jumbo', str(idx))
+            try:
+                src = {'Bucket': TTV_THUMB_BUCKET, 'Key': thumb['Key']}
+                dst = {
```

Comment:
It looks like we are not ever using `dst` itself, just the bucket and the key inside it. Can you replace with separate variables for the both of them?

Reply:
✅